// @visual-regression-tracker/agent-cypress/dist/commands.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.addVrtTrackBase64Command = exports.addVrtTrackBufferCommand = exports.addVrtTrackCommand = exports.addVrtStopCommand = exports.addVrtStartCommand = void 0;
/* global Cypress, cy */
var utils_1 = require("./utils");
var addVrtStartCommand = function () {
    Cypress.Commands.add("vrtStart", {
        prevSubject: ["optional"],
    }, function () {
        cy.task("VRT_START", {}, { log: false })
            .then(utils_1.handleError)
            .then(function () { return (0, utils_1.log)("Started"); });
    });
};
exports.addVrtStartCommand = addVrtStartCommand;
var addVrtStopCommand = function () {
    Cypress.Commands.add("vrtStop", {
        prevSubject: ["optional"],
    }, function () {
        cy.task("VRT_STOP", {}, { log: false })
            .then(utils_1.handleError)
            .then(function () { return (0, utils_1.log)("Stopped"); });
    });
};
exports.addVrtStopCommand = addVrtStopCommand;
var addVrtTrackCommand = function () {
    return Cypress.Commands.add("vrtTrack", {
        prevSubject: ["optional", "element", "window", "document"],
    }, function (subject, name, options, errorCallback) {
        (0, utils_1.trackImage)(subject, name, options).then(function (result) {
            return (0, utils_1.checkResult)(result, errorCallback);
        });
    });
};
exports.addVrtTrackCommand = addVrtTrackCommand;
var addVrtTrackBufferCommand = function () {
    return Cypress.Commands.add("vrtTrackBuffer", {
        prevSubject: false,
    }, function (name, imageBuffer, options) {
        (0, utils_1.trackBuffer)(name, imageBuffer, options).then(function (result) {
            return (0, utils_1.checkResult)(result);
        });
    });
};
exports.addVrtTrackBufferCommand = addVrtTrackBufferCommand;
var addVrtTrackBase64Command = function () {
    return Cypress.Commands.add("vrtTrackBase64", {
        prevSubject: false,
    }, function (name, imageBase64, options) {
        (0, utils_1.trackBase64)(name, imageBase64, options).then(function (result) {
            return (0, utils_1.checkResult)(result);
        });
    });
};
exports.addVrtTrackBase64Command = addVrtTrackBase64Command;
